Src/AutoNSubstituteUnitTest/SubstituteAttributeRelayTest.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Reflection;
+using NSubstitute;
+using Ploeh.AutoFixture.Kernel;
+using Ploeh.TestTypeFoundation;
+using Xunit;
+
+namespace Ploeh.AutoFixture.AutoNSubstitute.UnitTest
+{
+    public class SubstituteAttributeRelayTest
+    {
+        [Fact]
+        public void SutIsSpecimenBuilderToServeAsFixtureCustomization()
+        {
+            // Fixture setup
+            // Exercise system
+            // Verify outcome
+            Assert.True(typeof(ISpecimenBuilder).IsAssignableFrom(typeof(SubstituteAttributeRelay)));
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNull()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(null, context);
+            // Verify outcome
+            var expected = new NoSpecimen(null);
+            Assert.Equal(expected, specimen);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNotICustomAttributeProvider()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = new object();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(request, context);
+            // Verify outcome
+            var expected = new NoSpecimen(request);
+            Assert.Equal(expected, specimen);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenICustomAttributeProviderDoesNotReturnExpectedAttributeType()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = Substitute.For<ICustomAttributeProvider>();
+            request.GetCustomAttributes(Arg.Any<Type>(), Arg.Any<bool>()).Returns(new object[0]);
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(request, context);
+            // Verify outcome
+            var expected = new NoSpecimen(request);
+            Assert.Equal(expected, specimen);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateGetsInheritedAttributesFromProviderBecauseAttributesAreInheritedByDefault()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = Substitute.For<ICustomAttributeProvider>();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            sut.Create(request, context);
+            // Verify outcome
+            request.Received().GetCustomAttributes(typeof(SubstituteAttribute), true);

```

Comment:
The above test verifies `request` as though the return value doesn't matter. [If a test doesn't care about the return value of a function, then the reason for verifying it must be because calling the method has a side-effect that the test cares about.](http://blog.ploeh.dk/2013/10/23/mocks-for-commands-stubs-for-queries)